lxd/warnings.go

```diff
@@ -0,0 +1,382 @@
+//         metadata:
+//           $ref: "#/definitions/Warning"
+//   "404":
+//     $ref: "#/responses/NotFound"
+//   "500":
+//     $ref: "#/responses/InternalServerError"
+func warningGet(d *Daemon, r *http.Request) response.Response {
+	id := mux.Vars(r)["id"]
+
+	var dbWarning *db.Warning
+	var err error
+
+	err = d.cluster.Transaction(func(tx *db.ClusterTx) error {
+		dbWarning, err = tx.GetWarning(id)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	resp, err := dbWarningToAPIWarning(d, *dbWarning)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	return response.SyncResponse(true, resp)
+}
+
+// swagger:operation PUT /1.0/warnings/{uuid} warnings warning_put
+//
+// Update the warning
+//
+// Updates the warning status.
+//
+// ---
+// consumes:
+//   - application/json
+// produces:
+//   - application/json
+// parameters:
+//   - in: body
+//     name: warning
+//     description: Warning status
+//     required: true
+//     schema:
+//       $ref: "#/definitions/WarningPut"
+// responses:
+//   "200":
+//     $ref: "#/responses/EmptySyncResponse"
+//   "400":
+//     $ref: "#/responses/BadRequest"
+//   "403":
+//     $ref: "#/responses/Forbidden"
+//   "500":
+//     $ref: "#/responses/InternalServerError"
+func warningPut(d *Daemon, r *http.Request) response.Response {
+	id := mux.Vars(r)["id"]
+
+	req := api.WarningPut{}
+
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return response.BadRequest(err)
+	}
+
+	// Currently, we only allow changing the status to acknowledged or new.
+	status, ok := db.WarningTypes[req.Status]
+	if !ok {
+		// Invalid status
+		return response.BadRequest(fmt.Errorf("Invalid warning type %q", req.Status))
+	}
+
+	if status != db.WarningStatusAcknowledged && status != db.WarningStatusNew {
+		return response.Forbidden(fmt.Errorf(`Status may only be set to "acknowledge" or "new"`))
+	}
+
+	err = d.cluster.Transaction(func(tx *db.ClusterTx) error {
+		err := tx.UpdateWarningStatus(id, status)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	return response.EmptySyncResponse
+}
+
+// swagger:operation DELETE /1.0/warnings/{uuid} warnings warning_delete
+//
+// Delete the warning
+//
+// Removes the warning.
+//
+// ---
+// produces:
+//   - application/json
+// responses:
+//   "200":
+//     $ref: "#/responses/EmptySyncResponse"
+//   "500":
+//     $ref: "#/responses/InternalServerError"
+func warningDelete(d *Daemon, r *http.Request) response.Response {
+	id := mux.Vars(r)["id"]
+
+	err := d.cluster.Transaction(func(tx *db.ClusterTx) error {
+		err := tx.DeleteWarning(id)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	return response.EmptySyncResponse
+}
+
+func dbWarningToAPIWarning(d *Daemon, warning db.Warning) (api.Warning, error) {
```

Comment:
@monstermunchkin this feels like it should be a `ToAPI()` function on the `db.Warning` type (there's previous historic precedent for that with NodeInfo and Certificate).